File: src/main/java/com/haier/uhome/plugins/checker/ProjectChecker.java
package com.haier.uhome.plugins.checker;

import com.haier.uhome.plugins.utils.Utils;

import java.io.File;
import java.util.Arrays;
import java.util.List;

public class ProjectChecker implements IChecker {

    private final List<String> flutterCheckFiles = Arrays.asList("lib", ".packages", "pubspec.yaml");
    private final List<String> postGetCheckFiles = Arrays.asList("project_config", "postget.sh");
    private final List<String> gradleCheckFiles = Arrays.asList(".gradle", "gradle", "build.gradle",
            "gradlew", "gradlew.bat", "settings.gradle");
    private final List<String> buildCheckFiles = Arrays.asList("build.gradle");

    @Override
    public boolean checkFlutter(String path) {
        if (Utils.isEmptyString(path)) {
            return false;
        }
        File dir = new File(path);
        if (!dir.exists() || !dir.isDirectory() || dir.listFiles() == null) {
            return false;
        }
        int count = 0;
        for (File file : dir.listFiles()) {
            if (flutterCheckFiles.contains(file.getName())) {
                count++;
            }
        }
        return count == flutterCheckFiles.size();
    }

    @Override
    public boolean checkPostGet(String path) {
        if (Utils.isEmptyString(path)) {
            return false;
        }
        File dir = new File(path);
        if (!dir.exists() || !dir.isDirectory() || dir.listFiles() == null) {
            return false;
        }
        int count = 0;
        for (File file : dir.listFiles()) {
            if (postGetCheckFiles.contains(file.getName())) {
                count++;
            }
        }
        return count == postGetCheckFiles.size();
    }

    @Override
    public boolean checkGradle(String path) {
        if (Utils.isEmptyString(path)) {
            return false;
        }
        File dir = new File(path);
        if (!dir.exists() || !dir.isDirectory() || dir.listFiles() == null) {
            return false;
        }
        int count = 0;
        for (File file : dir.listFiles()) {
            if (gradleCheckFiles.contains(file.getName())) {
                count++;
            }
        }
        return count == gradleCheckFiles.size();
    }

    @Override
    public boolean checkBuildFile(String path) {
        if (Utils.isEmptyString(path)) {
            return false;
        }
        File dir = new File(path);
        if (!dir.exists() || !dir.isDirectory() || dir.listFiles() == null) {
            return false;
        }
        int count = 0;
        for (File file : dir.listFiles()) {
            if (buildCheckFiles.contains(file.getName())) {
                count++;
            }
        }
        return count == buildCheckFiles.size();
    }
}

File: src/main/kotlin/com/zcy/plugins/checker/ProjectChecker.kt
package com.zcy.plugins.checker

import com.intellij.openapi.project.Project
import com.zcy.plugins.utils.Utils
import java.io.File

class ProjectChecker : IChecker {

    val flutterCheckFiles = listOf("lib", ".packages", "pubspec.yaml")
    val postGetCheckFiles = listOf("project_config", "postget.sh")
    val gitCheckFiles = listOf(".git")

    /**
     * 检查当前路径是否为flutter工程
     *
     * @param path 待检查路径
     * @return 校验结果，true 为 通过
     */
    override fun checkFlutter(path: String?): Boolean {
        return check(path, flutterCheckFiles)
    }

    fun check(path: String?, checkFiles: List<String>): Boolean {
        if (Utils.isEmptyString(path)) {
            return false
        }
        val dir = File(path)
        if (!dir.exists() || !dir.isDirectory || null == dir.listFiles()) {
            return false
        }
        var count = 0
        dir.listFiles().forEach { file ->
            if (checkFiles.contains(file.name)) {
                count++
            }
        }
        return checkFiles.size == count
    }

    /**
     * 检查当前路径是否包含project_config信息，仅支持syn_module工程
     *
     * @param path 待检查路径
     * @return 校验结果，true 为 通过
     */
    override fun checkPostGet(path: String?): Boolean {
        return check(path, postGetCheckFiles)
    }


    /**
     * 检查当前打开的工作目录是否包含.git目录
     *
     * @param project 打开的工作目录
     * @return 校验结果，true 为 通过
     */
    override fun checkoutGit(project: Project?): Boolean {
        val dir = File(project!!.basePath)
        if (!dir.exists() || !dir.isDirectory || dir.listFiles() == null) {
            return false
        }
        var count = 0
        for (file in dir.listFiles()) {
            if (gitCheckFiles.contains(file.name)) {
                count++
            }
        }
        return count == gitCheckFiles.size
    }
}